mtwitter.js

```diff
@@ -0,0 +1,291 @@
+var Twitter = require('twitter');
+var Request = require('request');
+var fs = require('fs');
+var Buffer = require('buffer').Buffer;
+
+const TWITTER_PROFILE_INTERVAL = 60000;//ms;
+
+var MTwitter = function (bridge, config) {
+  this.app_auth = config.app_auth;
+  this.app_twitter = null;
+  this.tuser_cache = {};
+  this.timeline_list = [];
+  this.timeline_queue = [];
+  this.timeline_period = 0;
+  this.timeline_intervalID = null;
+  this.msg_queue = [];
+  this._bridge = bridge;
+
+
+  this.get_bearer_token().then((token) => {
+    this.app_auth.bearer_token = token;
+    console.log('Retrieved token');
+    this.app_twitter = new Twitter(this.app_auth);
+    setInterval(() => { this._process_queue(); }, 500);
+    this.start_timeline();
+  }).catch((error) => {
+      console.error('Error trying to retrieve bearer token:', error);
+      throw "Couldn't get a bearer token for Twitter AS.";
+  });
+};
+
+MTwitter.prototype._get_bearer_http = function () {
+  return new Promise( (resolve,reject) => {
+    var key = this.app_auth.consumer_key + ":" + this.app_auth.consumer_secret;
+    key = Buffer.from(key, 'ascii').toString('base64');
+    var options = {
+        url: "https://api.twitter.com/oauth2/token",
+        headers: {
+            'Authorization': "Basic " + key
+        },
+        form: "grant_type=client_credentials",
+        contentType: "application/x-www-form-urlencoded;charset=UTF-8"
+    };
+    Request.post(options, function(error, response, body) {
+        if (error) {
+            reject(error);
+        } else {
+            try {
+                var jsonresponse = JSON.parse(body);
```

Comment:
>This applies to basically every Twitter API call you're making.

I disagree with this for other calls. The node-twitter library will flag errors for [anything other than a 200](https://github.com/desmondmorris/node-twitter/blob/master/lib/twitter.js#L154) so I don't see the reason to check the error myself. This also goes for the JSON being valid, which is also checked.

I agree for this call I should be more careful however.